tapdb/asset_minting.go

```diff
@@ -891,11 +893,66 @@ func fetchAssetSprouts(ctx context.Context, q PendingAssetStore,
 		assetSprouts[i] = assetSprout
 	}
 
-	tapCommitment, err := commitment.FromAssets(nil, assetSprouts...)
+	// Construct a TapCommitment from the batch sprouts, and verify that the
+	// version is correct by recomputing the genesis output script.
+	tapCommitment, err := commitment.FromAssets(
+		fn.Ptr(commitment.TapCommitmentV2), assetSprouts...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// If we don't have any sprouts yet, it doesn't make sense to verify the
+	// commitment.
+	if len(assetSprouts) == 0 {
```

Comment:
Can lift this above, don't need to make a commitment at all if no sprouts...?

Reply:
Good point..I think `len(assetSprouts) == 0` should actually be an error here?

Reply:
Resolved this by updating the test to account for commitment root validation + moving this check earlier.